src/Core/Domain/Product/ValueObject/Price/RetailPrice.php

```diff
@@ -0,0 +1,104 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Domain\Product\ValueObject\Price;
+
+use PrestaShop\Decimal\Number;
+use PrestaShop\PrestaShop\Core\Domain\Exception\DomainConstraintException;
+use PrestaShop\PrestaShop\Core\Domain\Product\Exception\ProductConstraintException;
+use PrestaShop\PrestaShop\Core\Domain\TaxRule\Exception\TaxRuleConstraintException;
+use PrestaShop\PrestaShop\Core\Domain\TaxRule\ValueObject\TaxRuleId;
+use PrestaShop\PrestaShop\Core\Domain\ValueObject\Price;
+
+/**
+ * This is the net sales price for your customers.
+ * The retail price will automatically be calculated using the applied tax rate.
+ */
+final class RetailPrice
+{
+    /**
+     * @var bool
+     */
+    private $displayOnSaleFlag;
```

Comment:
Does it make sense to have this flag as part of `RetailPrice`?

Reply:
how do you suggest to pass it differently? Its related with RetailPrice

Reply:
For me it seems like a separate Product attribute that does not have anything to do with Retail Price. :thinking:

Reply:
same question https://github.com/PrestaShop/PrestaShop/pull/14824#discussion_r310132011